lgc/patch/Patch.cpp

```diff
@@ -219,9 +222,22 @@ void Patch::addPasses(PipelineState *pipelineState, legacy::PassManager &passMgr
 //
 // @param [in/out] passMgr : Pass manager to add passes to
 void Patch::addOptimizationPasses(legacy::PassManager &passMgr) {
+  // Use CodeGenOpt::Aggressive (i.e. 3) by default as this is what we have been
+  // doing so far.
+  CodeGenOpt::Level OLvl = CodeGenOpt::Aggressive;
+  switch (cl::OptLevel) {
+  default:
+    WithColor::error(errs(), "llpc") << "invalid optimization level.\n";
+    return;
+  case ' ': break;
+  case '0': OLvl = CodeGenOpt::None; break;
+  case '2': OLvl = CodeGenOpt::Aggressive; break;
+  case 'c': OLvl = CodeGenOpt::None; break; // -O0 for now
```

Comment:
What is -Oc to be, and should -O2 be -O3?

Reply:
-Oc is the new flag for compile-time tuning. I can certainly make this more clear in the cl::opt description above. Regarding -O2 vs -O3, the reason I chose -O2 is because currently when we create the TargetMachine (in LgcContext::Create) we use CodeGenOpt::Default (i.e. 2).

Reply:
I'd expect O1 to be the fast to compile optimization level.
But maybe it makes more sense to move further away from standard C/C++ optimization levels have something more descriptive, e.g., `--opt=[fast/optimized/min-vgprs]`, etc. These are probably not the best names, but more descriptive than `-O0/O2/Oc` IMO.

Reply:
I'd expect the -O option to be a reflection of the amount of optimisation that is being done - so I'd vote for more standard levels.
I agree that the fast option should probably be -O1, with -O0 being no opts (or at least, only required ones - which are arguably not opts :) )

For other factors - like min-vgpr, I'd expect that to be a separate option. You might well get a better min-vgpr with -O2 than with -O1 for instance, since we're telling the compiler to do more work at compile time.